File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.rb
# @param {String[]} words
# @param {Integer[]} groups
# @return {String[]}
def get_longest_subsequence(words, groups)
    return words if words.length == 1

    subsequence = []
    words.each_with_index do |word, i|
        if i == 0
            subsequence.push(word)
            next
        end

        subsequence.push(word) if groups[i] != groups[i - 1]
    end

    return subsequence
end

File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.go
func getLongestSubsequence(words []string, groups []int) []string {
    if len(words) == 1 {
        return words
    }

    subsequence := []string{}
    for i, word := range words {
        if i == 0 {
            subsequence = append(subsequence, word)
            continue
        }

        if groups[i] != groups[i-1] {
            subsequence = append(subsequence, word)
        }
    }

    return subsequence
}